### PR TITLE
core/types: transaction conditional KnownAccounts fix && HexOrDecimal deser

### PR DESCRIPTION
**Pull request by hamdiallam**
KnownAccounts should not use a pointer receiver for `MarshalJSON`. We can also leverage the `math.HexOrDecimal` to be more lenient when deserializing the conditional number values

### CODE REVIEW

File: core/types/transaction_conditional.go
Comment (hamdiallam) [inline core/types/transaction_conditional.go:45]: bugfix

**Review by vdamle — APPROVED** [review]

File: core/types/gen_transaction_conditional_json.go
Comment (tynes) [inline core/types/gen_transaction_conditional_json.go:35]: There should be a way to have the autogen tool output this. We shouldn't merge a diff to an autogenerated file.

Update the following struct:
```
type transactionConditionalMarshalling struct {
	BlockNumberMax *hexutil.Big
	BlockNumberMin *hexutil.Big
	TimestampMin   *hexutil.Uint64
	TimestampMax   *hexutil.Uint64
}
```

Like so:

```
type transactionConditionalMarshalling struct {
	BlockNumberMax *math.HexOrDecimal256
	BlockNumberMin *math.HexOrDecimal256
	TimestampMin   *math.HexOrDecimal256
	TimestampMax   *math.HexOrDecimal256
}
```

**Review by tynes — CHANGES_REQUESTED** [review]
Please fix and squash into a single commit with a clean commit message that follows the style of geth before merge

File: core/types/gen_transaction_conditional_json.go
Comment (hamdiallam) [reply core/types/gen_transaction_conditional_json.go:35]: ah yea good point! updated

**Comment by hamdiallam** [comment]
@tynes updated! Can you lmk what you mean by geth style commit message? I think the title reflects what I see in the geth PRs, `package: description`, and there doesn't seem to be a format for the description

**Comment by tynes** [comment]
> @tynes updated! Can you lmk what you mean by geth style commit message? I think the title reflects what I see in the geth PRs, `package: description`, and there doesn't seem to be a format for the description

The title of the PR approximates a geth commit message, like you said `package name: description`. I notice that your commit message now doesn't follow that but the PR title does instead? We try to keep the git history of `op-geth` very clean so that its possible to easily view the diff over time.

I also think its a bad idea to manually modify autogenerated files, I notice that the allow unused fields diff is lost in this PR, I missed that in review previously and wouldn't have approved that if I noticed it at the time

**Comment by hamdiallam** [comment]
> > @tynes updated! Can you lmk what you mean by geth style commit message? I think the title reflects what I see in the geth PRs, `package: description`, and there doesn't seem to be a format for the description
> 
> The title of the PR approximates a geth commit message, like you said `package name: description`. I notice that your commit message now doesn't follow that but the PR title does instead? We try to keep the git history of `op-geth` very clean so that its possible to easily view the diff over time.
> 
> I also think its a bad idea to manually modify autogenerated files, I notice that the allow unused fields diff is lost in this PR, I missed that in review previously and wouldn't have approved that if I noticed it at the time

Ahh wait I think I misunderstood. Didn't realize the commit message wasn't just the PR title. Renamed the commit title to match

And noted on the generated code. The generator doesn't seem to have an option for unknown fields but that's not super important anyways

**Review by tynes — APPROVED** [review]

**Pull request merged**
